File: euler_project/0004/p004.py
# def is_palindrome(n):
#        str(n) == str(n)[::-1]

def solve():	
	ans = 0;
	for x in range(100, 1000):
		for y in range(100, 1000):
			s = str(x * y);
			ls = list(s)
			ls.reverse()
			if (list(s) == ls and x * y > ans):
				ans = x * y
	return ans

print solve()

File: euler_project/p004/p004.erl
%%%-------------------------------------------------------------------
%%% @author xuexg
%%% @doc
%%%
%%% @end
%%% Created : 17. 十月 2017 15:42
%%%-------------------------------------------------------------------
-module(p004).

%% API
-export([solve/0]).

solve() ->
    PalindromeList = [X * Y ||
        X <- lists:seq(100, 999),
        Y <- lists:seq(100, 999),
        integer_to_list(X * Y) =:= lists:reverse(integer_to_list(X * Y))
        ],
    io:format("The answer is ~p~n", [lists:max(PalindromeList)]).

%% erlang shell run: p004:solve() = 906609